From a22629e0c6878d6feea97a16b550bb24e85729e4 Mon Sep 17 00:00:00 2001
From: Damien Lacoste <damien.lacoste@esrf.fr>
Date: Wed, 23 Nov 2022 10:22:10 +0100
Subject: [PATCH] Add define in AttributeTraits to be backward compatible with
 spdlog 1.9.2

Note that as spdlog version 1.11 becomes mainstream in distributions we
should update the code to use the new api as this fix will be dropped in
new versions of the fmt library on which spdlog rely.
---
 src/AttributeTraits.hpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/AttributeTraits.hpp b/src/AttributeTraits.hpp
index 5e0a87f..9ecb323 100644
--- a/src/AttributeTraits.hpp
+++ b/src/AttributeTraits.hpp
@@ -20,6 +20,14 @@
 #ifndef _ATTRIBUTE_TRAITS_H
 #define _ATTRIBUTE_TRAITS_H
 
+// See https://github.com/fmtlib/fmt/releases/tag/9.0.0
+// To support spdlog 1.11 we need to use this define
+// otherwise we need to update the print(ostream&) method
+// to use the new api.
+// Since we still want to support spdlog 1.9.2 (default version in ubuntu 22.04)
+// we use this define for now
+#define FMT_DEPRECATED_OSTREAM
+
 #include "LibUtils.hpp"
 
 #include <iostream>
-- 
2.38.1

